huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go

```diff
@@ -234,50 +256,48 @@ func ResourceAlarmRule() *schema.Resource {
 	}
 }
 
-func getMetricOpts(d *schema.ResourceData) (alarmrule.MetricOpts, error) {
-	mos, ok := d.Get("metric").([]interface{})
-	if !ok {
-		return alarmrule.MetricOpts{}, fmtp.Errorf("Error converting opt of metric:%v", d.Get("metric"))
-	}
-	mo := mos[0].(map[string]interface{})
-
-	mod := mo["dimensions"].([]interface{})
-	dopts := make([]alarmrule.DimensionOpts, len(mod))
-	for i, v := range mod {
-		v1 := v.(map[string]interface{})
-		dopts[i] = alarmrule.DimensionOpts{
-			Name:  v1["name"].(string),
-			Value: v1["value"].(string),
+func getMetricOpts(d *schema.ResourceData) alarmrule.MetricOpts {
+	metricRaw := d.Get("metric").([]interface{})
+	metric := metricRaw[0].(map[string]interface{})
+
+	dimensionsRaw := metric["dimensions"].([]interface{})
+	dimensionsOpts := make([]alarmrule.DimensionOpts, len(dimensionsRaw))
+	for i, dimensionRaw := range dimensionsRaw {
+		dimension := dimensionRaw.(map[string]interface{})
+		dimensionsOpts[i] = alarmrule.DimensionOpts{
+			Name:  dimension["name"].(string),
+			Value: dimension["value"].(string),
 		}
 	}
+
 	return alarmrule.MetricOpts{
-		Namespace:  mo["namespace"].(string),
-		MetricName: mo["metric_name"].(string),
-		Dimensions: dopts,
-	}, nil
+		Namespace:  metric["namespace"].(string),
+		MetricName: metric["metric_name"].(string),
+		Dimensions: dimensionsOpts,
+	}
 }
 
 func getAlarmAction(d *schema.ResourceData, name string) []alarmrule.ActionOpts {
-	aos := d.Get(name).([]interface{})
-	if len(aos) == 0 {
-		return nil
-	}
-	opts := make([]alarmrule.ActionOpts, len(aos))
-	for i, v := range aos {
-		v1 := v.(map[string]interface{})
-
-		v2 := v1["notification_list"].([]interface{})
-		nl := make([]string, len(v2))
-		for j, v3 := range v2 {
-			nl[j] = v3.(string)
-		}
+	if v, ok := d.GetOk(name); ok {
+		actionOptsRaw := v.([]interface{})
+		actionOpts := make([]alarmrule.ActionOpts, len(actionOptsRaw))
+		for i, actionOptRaw := range actionOptsRaw {
+			actionOpt := actionOptRaw.(map[string]interface{})
+
+			notificationListRaw := actionOpt["notification_list"].([]interface{})
+			notificationList := make([]string, len(notificationListRaw))
+			for j, notification := range notificationListRaw {
+				notificationList[j] = notification.(string)
+			}
 
-		opts[i] = alarmrule.ActionOpts{
-			Type:             v1["type"].(string),
-			NotificationList: nl,
+			actionOpts[i] = alarmrule.ActionOpts{
+				Type:             actionOpt["type"].(string),
+				NotificationList: notificationList,
+			}
 		}
+		return actionOpts
 	}
-	return opts
+	return nil
 }
 
 func getAlarmCondition(d *schema.ResourceData) alarmrule.ConditionOpts {
```

Comment:
ditto.

Reply:
func name updated, changing argument not adopted